compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/expressions/CheckExpressionTest.java

```diff
@@ -47,4 +67,34 @@ public void testCheckWithMissingExpr() {
     public void testCheckWithMissingBinaryOp() {
         testFile("check-expr/check_expr_source_04.bal", "check-expr/check_expr_assert_04.json");
     }
+
+    @Test
+    public void testOnFailCheckWithMissingFail() {
+        testFile("on-fail-check/on_fail_check_source_02.bal", "on-fail-check/on_fail_check_assert_02.json");
```

Comment:
Shall we move these tests to `check-expr` directroy? We keep one directory for each test class. Since this is CheckExpressionTest class let's add the new tests under the same directory. We can use the name (`on_fail_check_`) to distinguish.